File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        
        Arrays.sort(phone_book);
        for(int i=0; i<phone_book.length-1; i++){
            boolean flag = true;
            for(int j=0; j<phone_book[i].length(); j++){
                if(phone_book[i].charAt(j)!=phone_book[i+1].charAt(j)){
                    flag = false;
                    break;
                }
            }
            if(flag) return false;
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

bool solution(vector<string> phone_book) { 
    bool answer = true;
    sort(phone_book.begin(), phone_book.end());
    for(int i=0; i<phone_book.size()-1; i++){
        string a = phone_book[i];
        string b = phone_book[i+1];
        
        if(a.size()>b.size()) continue;
        
        for(int j=0; j<a.size(); j++){
            if(j==a.size()-1 && a[j]==b[j]) answer = false;
            else if(a[j]!=b[j]) break;
        }
    }
    return answer;
}